fighters/common/src/function_hooks/change_status.rs

```diff
@@ -6,11 +6,37 @@ unsafe fn change_status_request_hook(boma: &mut BattleObjectModuleAccessor, stat
     let mut next_status = status_kind;
 
     if boma.is_fighter() {
-        if StatusModule::status_kind(boma) == *FIGHTER_STATUS_KIND_AIR_LASSO_REWIND
-        && (next_status == *FIGHTER_STATUS_KIND_CLIFF_CATCH || next_status == *FIGHTER_STATUS_KIND_CLIFF_CATCH_MOVE || next_status == *FIGHTER_STATUS_KIND_CLIFF_WAIT)
-        &&  VarModule::is_flag(boma.object(), vars::common::instance::SHOULD_TRUMP_TETHER) {
-            VarModule::off_flag(boma.object(), vars::common::instance::SHOULD_TRUMP_TETHER);
-            next_status = *FIGHTER_STATUS_KIND_CLIFF_ROBBED;
+        // Tether trump logic
+        if boma.is_status(*FIGHTER_STATUS_KIND_AIR_LASSO_REWIND)
+        && [*FIGHTER_STATUS_KIND_CLIFF_CATCH, *FIGHTER_STATUS_KIND_CLIFF_CATCH_MOVE, *FIGHTER_STATUS_KIND_CLIFF_WAIT].contains(&next_status) {
+            let player_number = WorkModule::get_int(boma, *FIGHTER_INSTANCE_WORK_ID_INT_ENTRY_ID) as u32;
+            let pos = GroundModule::hang_cliff_pos_3f(boma);
+
+            for i in 0..8 {
+                if let Some(object_id) = ::utils::util::get_active_battle_object_id_from_entry_id(i) {
+                    let object = ::utils::util::get_battle_object_from_id(object_id);
+                    if !object.is_null() {
+                        if i == player_number || VarModule::get_float(object, vars::common::instance::LEDGE_POS_X) == 0.0 {
+                            continue;
+                        }
+    
+                        if pos.x == VarModule::get_float(object, vars::common::instance::LEDGE_POS_X) && pos.y == VarModule::get_float(object, vars::common::instance::LEDGE_POS_Y) {
+                            next_status = *FIGHTER_STATUS_KIND_CLIFF_ROBBED;
+                        }
+    
+                        let module_accessor = &mut *(*object).module_accessor;
+                        if module_accessor.kind() == *FIGHTER_KIND_POPO {
```

Comment:
I have a single concern and it's about this line.

I believe the fighter check should be for either POPO or NANA, because there are certain costumes where Nana is the one in the lead. However, I can't verify this on my own, so I ask that you please check.

Reply:
I have been told that the lead always reads as POPO